lambdas/evaluate-gpg45-scores/src/main/java/uk/gov/di/ipv/core/evaluategpg45scores/EvaluateGpg45ScoresHandler.java

```diff
@@ -131,4 +164,80 @@ private List<ContraIndicatorItem> getCIsAndSwallowErrors(
             return List.of();
         }
     }
+
+    @Tracing
+    private void updateSuccessfulVcStatuses(IpvSessionItem ipvSessionItem, List<String> credentials)
+            throws ParseException {
+
+        // get list of success vc's
+        List<VcStatusDto> currentVcStatusDtos = ipvSessionItem.getCurrentVcStatuses();
+
+        if (currentVcStatusDtos == null) {
+            currentVcStatusDtos = new ArrayList<>();
+        }
+
+        if (currentVcStatusDtos.size() != credentials.size()) {
+            List<VcStatusDto> updatedStatuses = generateVcSuccessStatuses(credentials);
+            ipvSessionItem.setCurrentVcStatuses(updatedStatuses);
+            ipvSessionService.updateIpvSession(ipvSessionItem);
+        }
+    }
+
+    private List<VcStatusDto> generateVcSuccessStatuses(List<String> credentials)
+            throws ParseException {
+        List<VcStatusDto> vcStatuses = new ArrayList<>();
+
+        for (String credential : credentials) {
+            SignedJWT signedJWT = SignedJWT.parse(credential);
+            JSONObject vcClaim = (JSONObject) signedJWT.getJWTClaimsSet().getClaim(VC_CLAIM);
+            JSONArray evidenceArray = (JSONArray) vcClaim.get(VC_EVIDENCE);
+            if (evidenceArray == null) {
+                CredentialIssuerConfig addressCriConfig =
+                        configurationService.getCredentialIssuer(addressCriId);
+                String vcIss = signedJWT.getJWTClaimsSet().getIssuer();
+                if (vcIss.equals(addressCriConfig.getAudienceForClients())) {
+                    vcStatuses.add(new VcStatusDto(vcIss, true));
+                }
```

Comment:
Is it worth logging something here if they don't match? Feels like a situation we should never be in - having a VC with no evidence from a CRI other than the address CRI?

Reply:
Yeah good shout 👍  I'll add a warn log level here